### DISCUSSION TITLE
Project: happi - device database for ophyd (and other) devices

### DISCUSSION THREAD

**Discussion by klauer** [pcdshub/epics-codeathon-2022-python#10] [Projects]
Source code: https://github.com/pcdshub/happi/
Documentation: https://pcdshub.github.io/happi

> LCLS endstations deal with dynamic sets of instrumentation. Information like ports, triggers and aliases are all important for operation, but hard to manage when spread across a multitude of applications. Happi solves this problem by creating a single access point for all the metadata required to interface with LCLS instrumentation. Using a flexible container based system Happi allows the enforcement of specific conventions while still permitting flexible data entry where required.

Usage of happi is not limited to just the LCLS. It offers the ability to generically describe and instantiate your facility's Python devices (any object, really!).

Related:
* https://github.com/pcdshub/epics-codeathon-2022-python/discussions/8 (supports happi for device creation)

**Comment by klauer** [comment]
## Project idea

General issue:
* Introduce happi (and ophyd) to the community
* Sort through the issue list
* Reproduce the issue and make an attempt at a fix

## Pull Request / Issue link

Any open issue:
https://github.com/pcdshub/happi/issues

**Comment by klauer** [comment]
## Project idea

Improve documentation.
Starting point issues: 
* https://github.com/pcdshub/happi/issues/204
* And others: https://github.com/pcdshub/happi/issues?q=is%3Aopen+is%3Aissue+label%3ADocumentation

**Comment by klauer** [comment]
## Project idea

Fix conda recipe for optional dependencies
* https://github.com/pcdshub/happi/issues/202
* https://github.com/pcdshub/happi/blob/master/conda-recipe/meta.yaml
* Base the updated recipe on the one from conda-forge: https://github.com/conda-forge/happi-feedstock/blob/main/recipe/meta.yaml

**Reply by tangkong** [reply]
https://github.com/pcdshub/happi/pull/240

**Comment by klauer** [comment]
## Project idea

Fix up command-line interface.

Issues:
* https://github.com/pcdshub/happi/issues?q=is%3Aopen+label%3ACLI+label%3A%22help+wanted%22

**Comment by klauer** [comment]
## Project idea

Add `enforce_doc` keyword argument to explain individual aspects of happi item components.

Issue: https://github.com/pcdshub/happi/issues/217

**Reply by tangkong** [reply]
https://github.com/pcdshub/happi/pull/241
